### PR TITLE
change enumDependencies to enum for Product Taxonomy

### PR DESCRIPTION
**Pull request by fkpwolf**
[AB#389331](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/389331)

### CODE REVIEW

**Review by yufeih — APPROVED** [review]

File: src/docfx/config/ops/OpsMetadataRuleConverter.cs [hunk 1]
```diff
@@ -116,16 +109,38 @@ public static string GenerateJsonSchema(string rulesContent, string allowlistsCo
...
-                    schema.enumDependencies.Add($"{attribute}[0]", taxonomy);
-
-                    if (rulesInfo.TryGetValue("Match", out var matchRuleInfo) &&
-                        !string.IsNullOrEmpty(matchRuleInfo.Value) &&
-                        !schema.enumDependencies[$"{attribute}[0]"].ContainsKey(matchRuleInfo.Value))
+                    // if just plain enum, extend property
+                    if (enumValues != null && enumValues.Length > 0)
                     {
-                        schema.enumDependencies[$"{attribute}[0]"].Add(matchRuleInfo.Value, null);
+                        var enumType = new
```
Comment (OsmondJiang) [inline src/docfx/config/ops/OpsMetadataRuleConverter.cs:117]: do we need consider whether it's array or string? directly adding `iterms` seems risky

File: src/docfx/config/ops/OpsMetadataRuleConverter.cs [hunk 2]
```diff
@@ -163,27 +178,41 @@ private static bool TryGetAttributeCustomRules(Dictionary<string, OpsMetadataRul
...
             {
                 taxonomy = new Dictionary<string, EnumDependenciesSchema?>();
+                enumValues = null;
                 return false;
             }
 
             if (string.IsNullOrEmpty(subTaxonomy.NestedValue))
             {
                 taxonomy = subTaxonomy.NestedTaxonomy.list.ToDictionary(x => x, x => (EnumDependenciesSchema?)null);
+                enumValues = null;
```
Comment (OsmondJiang) [inline src/docfx/config/ops/OpsMetadataRuleConverter.cs:198]: can we also do the same thing for taxonomy of which the nestedValue is null?  like ms.topic

**Review by OsmondJiang — CHANGES_REQUESTED** [review]
to prevent the auto merge

File: src/docfx/config/ops/OpsMetadataRuleConverter.cs (hunk 1 above)
Comment (fkpwolf) [reply src/docfx/config/ops/OpsMetadataRuleConverter.cs]: ok

**Comment by fkpwolf** [comment]
the diff looks not relevant with this changes and I can't find diff details in `publish artifacts`. So I think this PR is ok for merging. @OsmondJiang

**Comment by OsmondJiang** [comment]
@yufeih @mingwli could you help confirm the diff(exceed 60s) is expected?

**Comment by fkpwolf** [comment]
Here are one success [build log](https://dev.azure.com/ceapex/Engineering/_build/results?buildId=372847&view=logs&j=43d08477-5396-50a3-fedd-32f526c9defc&t=7f91c047-fae8-5c9e-afc1-f6b0f36191b6) and [one failed log](https://dev.azure.com/ceapex/Engineering/_build/results?buildId=373511&view=logs&j=43d08477-5396-50a3-fedd-32f526c9defc&t=7f91c047-fae8-5c9e-afc1-f6b0f36191b6). One took 2m 11s, and one took 2m 16s.

For this pr, it will change output of converting rueset to schema.json. From above log, the diff is:
before this change:
```json
"properties":{
  "ms.topic": {
              "type": [
                  "string",
                  "null"
              ]
   }
 }

"enumDependencies": {
    "ms.topic[0]": {
            "archived": null,
            "article": null,
    }
}
```
After this change:
```json
"properties": {
  "ms.topic": {
              "type": [
                  "string",
                  "null"
              ],
              "enum": [
                  "archived",
                  "article",
                  "best-practice",
                  "callback",
             ]
  }
}
```
docfx added lots of `ms.topic` to Markdown metadata.
https://github.com/MicrosoftDocs/VBA-Docs/blob/999adf1015278d4354fcfcf4043ca619919ed162/docfx.json#L50-L63
So root cause of pipeline timeout would be validation of `enum` is slower than `enumDependencies`.

**Comment by yufeih** [comment]
This isn't blocking to me. `dryrun.azure-docs-pr` perf stats does not show up any json schema related hotspots.

**Review by OsmondJiang — APPROVED** [review]

**Comment by fkpwolf** [comment]
> This isn't blocking to me. `dryrun.azure-docs-pr` perf stats does not show up any json schema related hotspots.

I check code of json schema validator on enum and didn't find suspect point. If so, pls help to merge this pr.

**Pull request merged**